### PR TITLE
enable baud rate for serial feature to be updated

### PR DESCRIPTION
**Pull request by soundanalogous**
This is useful when you need to change the baud rate of a serial peripheral.

### CODE REVIEW

File: utility/SerialFirmata.cpp [hunk 1]
```diff
@@ -199,6 +199,23 @@ boolean SerialFirmata::handleSysex(byte command, byte argc, byte *argv)
         }
         break; // SERIAL_LISTEN
 #endif
+      case SERIAL_UPDATE_BAUD:
+        serialPort = getPortFromId(portId);
+        if (serialPort == NULL) {
+          break;
+        }
+        long newBaud = (long)argv[1] | ((long)argv[2] << 7) | ((long)argv[3] << 14);
+        if (portId < 8) {
```
Comment (soundanalogous) [inline utility/SerialFirmata.cpp]: consider calling `flush()` before calling `end()`

**Comment by dtex** [comment]
Jeff,

I'm sorry to do this to you but how do I update the Arduino IDE to use the source from the repo? You once sent me a zip (when testing something else) that I dropped into the app package (OS X) and I'm trying to do that with the repo, but I must be putting it in the wrong path. I even deleted the existing firmata folder at ```content/java/libraries/``` but firmata still shows up in examples. I'm using 1.6.5

**Comment by soundanalogous** [comment]
There is no need to copy Firmata into the Arduino IDE bundle as long as you are using Arduino IDE 1.6.4 or higher.

Simply copy (or clone) Firmata (renaming to `Firmata` when you clone) into you sketchbook libraries directory.

**Comment by dtex** [comment]
I tried many times and could not get a successful switch to 9600 (based on the data I'm getting back).

Here's my test code. As best I can tell it should behave exactly like [this example](https://github.com/firmata/arduino/issues/260#issuecomment-168435942). I added the timeouts just to be conservative:

````js
var firmata = require("firmata");
var board = new firmata("/dev/cu.usbmodem14231");

board.on("ready", function() {
  
  board.serialConfig({portId: board.SERIAL_PORT_IDs.HW_SERIAL1, baud: 115200});

  setTimeout( function() {
    board.serialRead(board.SERIAL_PORT_IDs.HW_SERIAL1, function(data) {
      console.log("read", data);
    });
  }, 1000);

  setTimeout( function() {
    board.serialWrite(board.SERIAL_PORT_IDs.HW_SERIAL1, [0xA0, 0xA2, 0x00, 0x18, 0x81,
      0x02, 0x01, 0x01, 0x00, 0x01, 0x01, 0x01, 0x05, 0x01, 0x01, 0x01,
      0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x00, 0x01, 0x25,
      0x80, 0x01, 0x3A, 0xB0, 0xB3]);
  }, 2000);

  setTimeout( function() {
    board.serialBaud(board.SERIAL_PORT_IDs.HW_SERIAL1, 9600);
  }, 3000);

});
````

**Comment by soundanalogous** [comment]
Did you check out the [update-baud branch](https://github.com/firmata/arduino/tree/update-baud) before uploading StandardFirmataPlus to your board?

**Comment by soundanalogous** [comment]
I know what the problem is. I made a really stupid mistake in the protocol. I can't use `0x80` (for `SERIAL_UPDATE_BAUD`) as a command byte because it's > 127 decimal. Luckily I left `0x00` open so I can use that to signal an extended command set. I should really start allocating 2 bytes for commands from now on. I'll have to go back to the protocol stage to fix this.

**Comment by soundanalogous** [comment]
Here's one solution: https://github.com/firmata/protocol/pull/46#issuecomment-172748196.

**Comment by soundanalogous** [comment]
I'm going to release a SerialFirmata 2.0 at some point that fixes the larger scalability issue here and will then revisit the ability to update the serial baud rate.

**Pull request closed without merging**